### PR TITLE
Profile

### PR DESCRIPTION
**Pull request by tpreusse**
Adding all fields for an awesome author page: bio, pgp key and singal phone number (with different read access options)

And migrated testimonial on to the user (image to be used everywhere, statement also being part of the profile and role to be credentials for comments).

### CODE REVIEW

File: graphql/resolvers/_mutations/updateMe.js [hunk 1]
```diff
@@ -1,61 +1,163 @@
...
-        },
+        { id: me.id },
+        updateFields.reduce(
+          (updates, key) => {
+            updates[key] = args[key]
+            return updates
+          },
+          {
+            portraitUrl
+          }
```
Comment (patte) [inline graphql/resolvers/_mutations/updateMe.js]: missing updatedAt

File: graphql/resolvers/_mutations/updateMe.js [hunk 2]
```diff
@@ -1,61 +1,163 @@
...
     }
     if (address) {
-      if (req.user.addressId) {
+      if (me._raw.addressId) {
         // update address of user
         await transaction.public.addresses.update(
-          { id: req.user.addressId },
+          { id: me._raw.addressId },
           address
         )
```
Comment (patte) [inline graphql/resolvers/_mutations/updateMe.js]: missing updatedAt

File: graphql/resolvers/_mutations/updateMe.js [hunk 3]
```diff
@@ -64,13 +166,14 @@ module.exports = async (_, args, { pgdb, req, user: me, t }) => {
...
         await transaction.public.users.updateOne(
-          { id: req.user.id },
+          { id: me.id },
           { addressId: userAddress.id }
```
Comment (patte) [inline graphql/resolvers/_mutations/updateMe.js]: missing updatedAt

File: graphql/resolvers/_queries/statements.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    const users = await pgdb.query(`
+      SELECT
+        u.*,
+        m."sequenceNumber" as "sequenceNumber"
+      FROM users u
+      JOIN memberships m
+        ON m.id = (SELECT id FROM memberships WHERE "userId" = u.id ORDER BY "sequenceNumber" ASC LIMIT 1) 
+      WHERE
+        ARRAY[u.id] && 
+        :ids;
```
Comment (patte) [inline graphql/resolvers/_queries/statements.js]: I would keep that on one line, it's confusing enough already

File: graphql/resolvers/_queries/statements.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+      WHERE
+        u."isListed" = true AND u."isAdminUnlisted" = false AND
+        (
+          u."firstName" % :search OR
+          u."lastName" % :search OR
+          u."firstName" ILIKE :searchLike OR
+          u."lastName" ILIKE :searchLike OR
+          c.description % :search OR
+          c.description ILIKE :searchLike
+        );
```
Comment (patte) [inline graphql/resolvers/_queries/statements.js:85]: this is not a valid sql query 🧐

**Review by patte — CHANGES_REQUESTED** [review]
very nice work!

File: graphql/resolvers/_queries/statements.js (hunk 5 above)
Comment (tpreusse) [reply graphql/resolvers/_queries/statements.js:85]: totally didn't finish that one, done now: https://github.com/orbiting/republik-backend/pull/25/commits/4452e984fe59f97d7440bcc955f8da50f83988d1

**Comment by patte** [comment]
could you please check if this has any implications on [decorateTree](https://github.com/orbiting/republik-backend/blob/master/graphql/resolvers/Discussion/comments.js#L134)?

**Comment by patte** [comment]
otherwise feel free to merge

**Review by patte — APPROVED** [review]

**Comment by patte** [comment]
can you tell what's the deal with hasPublicProfile and isListed?

**Comment by tpreusse** [comment]
`decorateTree` should be fine and not affected. But only tested on the surface. DisplayAuthor will remain and is unchanged

**Comment by tpreusse** [comment]
`hasPublicProfile` = fine with having a linkable profile (e.g. https://www.republik.ch/~tpreusse)
`isListed` = fine with being listed on `/community`

`credentials.isListed` = fine with being listed on profile and possible future lists (anonymous credentials shouldn't be auto added to profile)

**Pull request merged**